demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py

```diff
@@ -38,44 +38,70 @@ def generate_example_dict(examples_file: Optional[str], insecure=False):
+        command_name: the command name whose outputs we want to manipulate.
         output_with_contexts: the new outputs.
     """
-    commands = yml_data['script']['commands']
-    found = False
-    for cmd in commands:
-        if cmd.get('name') == command:
-            cmd['outputs'] = output_with_contexts
-            found = True
-            break
-
-    if not found:
-        raise Exception(
-            f'Input YML doesn\'t have the "{command}" command that exists in the examples file.')
-
+    commands = yml_data.get('script', {}).get('commands') or []
+    command_names = [command.get('name') for command in commands]
+    if command_name not in command_names:
+        raise ValueError(f'The {command_name} command is missing from the integration YML.')
+    command_index = command_names.index(command_name)
+    command = commands[command_index]
+
+    outputs: List[Dict[str, str]] = command.get('outputs') or []
+    old_descriptions = _output_path_to_description(outputs)
+    new_descriptions = _output_path_to_description(output_with_contexts)
+    old_output_paths = {output.get('contextPath') for output in command.get('outputs', [])}
+
+    outputs.extend(output for output in output_with_contexts
+                   if output.get('contextPath') and output.get('contextPath') not in old_output_paths)
+
+    # populates the description field, preferring the new value (if not blank), and existing values over blanks.
+    for output in outputs:
+        path = output.get('contextPath')
+        if not path:
+            raise ValueError('Found a command without a contextPath value')
+        if not output.get('description'):
+            output['description'] = new_descriptions.get(path) or old_descriptions.get(path) or ''
+    yml_data['script']['commands'][command_index]['outputs'] = outputs
     return yml_data
 
 
+def _output_path_to_description(command_outputs: List[Dict[str, str]]) -> Dict[str, str]:
+    """ creates a mapping of contextPath -> description, if the description is not null. """
+    descriptions = {}
+    for output in command_outputs:
+        description = output.get('description')
+        context_path = output.get('contextPath')
+        if context_path and description:
+            descriptions[context_path] = description
+    return descriptions
+
+
 def generate_integration_context(
         input_path: str,
         examples: Optional[str] = None,
         insecure: bool = False,
         verbose: bool = False,
+        output_path: Optional[str] = None
 ):
     """ Generate integration command contexts in-place.
 
     Args:
+        output_path: Output path
         input_path: path to the yaml integration.
         examples: path to the command examples.
         insecure: should use insecure.
         verbose: verbose (debug mode).
     """
-
+    if not output_path:
+        print_warning('output path wasn\'t provided, saving in input path')
```

Comment:
lets not do warning. lets just print "saving output to <path>" or something like that.